src/fury/encoder/field_info.h

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2023 The Fury Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include "fury/encoder/preprocessor.h"
```

Comment:
Seems `preprocessor.h` and `field_info` is not releated to row format, can we move it to a separate directory, so we can reuse it for object graph serialization in future

Reply:
Maybe put this under `src/fury` directly or a directory such as `meta/reflection`?